node/tests/utils.rs

```diff
@@ -471,6 +478,101 @@ impl MASQNode {
+                Some(next_string) => {
+                    if next_string.starts_with("--") {
+                        acc.push((current_string, None));
+                        acc
+                    } else {
+                        let value = peekable.next().unwrap();
+                        acc.push((current_string, Some(value)));
+                        acc
+                    }
+                }
+                None => {
+                    acc.push((current_string, None));
+                    acc
+                }
+            }
+        }
+
+        let count = args.len();
+        let mut peekable = args.into_iter().peekable();
+        (0..count).fold(vec![], |acc, _| {
+            match peekable.next() {
+                //arg name
+                Some(current_string) => {
+                    look_at_val_distanced_by_two(&mut peekable, current_string, acc)
+                }
+                None => acc,
+            }
+        })
+    }
+
+    fn extend_args_without_duplication(
+        default_args: Vec<String>,
+        config_args: Vec<String>,
+    ) -> Vec<String> {
+        fn retain_unique_default_args(
+            default_args: Vec<String>,
+            config_args: &BTreeMap<String, Option<String>>,
+        ) -> BTreeMap<String, Option<String>> {
+            MASQNode::virtual_arg_pairs(default_args)
+                .into_iter()
+                .flat_map(|(arg_name, value)| {
+                    let res = config_args
+                        .keys()
+                        .contains(&&arg_name)
+                        .not()
+                        .then_some((arg_name, value));
+                    res
+                })
+                .collect()
+        }
+        fn test_uniqueness_for_config_args(config_args_as_tuples: &[(String, Option<String>)]) {
+            let len_before = config_args_as_tuples.len();
+            let config_args_to_test: Vec<_> = config_args_as_tuples
+                .iter()
+                .sorted()
+                .dedup_by(|(arg_1, _), (arg_2, _)| arg_1 == arg_2)
+                .collect();
+            let len_after = config_args_to_test.len();
+            assert_eq!(
+                len_after, len_before,
+                "You supplied additional arguments with some of \
+                them duplicated, use each only once! {:?}",
+                config_args_as_tuples
+            );
+        }
+
+        //usages of BTreeMaps to preserve the order of arguments, both defaulted and from the config
+        let config_args_as_tuples = Self::virtual_arg_pairs(config_args);
+        test_uniqueness_for_config_args(&config_args_as_tuples);
+        let config_args: BTreeMap<String, Option<String>> =
+            BTreeMap::from_iter(config_args_as_tuples.into_iter());
+        let default_args_to_keep = retain_unique_default_args(default_args, &config_args);
+        default_args_to_keep
+            .into_iter()
+            .chain(config_args.into_iter().rev())
```

Comment:
This `rev()` may be unnecessary. If it's not unnecessary, perhaps a comment could explain it.

@bertllll said, "If it's necessary, move `rev()` closer to the `BTreeMap::from_iter()` on `config_args`."